filterparser.go

```diff
@@ -109,40 +110,60 @@ var operatorStart = map[rune]interface{}{
 	'!': nil,
 }
 
-var operatorsToToken = map[string]parserToken{
-	wordEQUAL:       parserTokenEQUAL,
-	wordNOTEQUAL:    parserTokenNOTEQUAL,
-	wordLT:          parserTokenLT,
-	wordLTE:         parserTokenLTE,
-	wordGT:          parserTokenGT,
-	wordGTE:         parserTokenGTE,
-	wordCONTAINS:    parserTokenCONTAINS,
-	wordNOTCONTAINS: parserTokenNOTCONTAINS,
-	wordMATCHES:     parserTokenMATCHES,
-	wordIN:          parserTokenIN,
-	wordNOTIN:       parserTokenNOTIN,
-	wordNOT:         parserTokenNOT,
-	wordEXISTS:      parserTokenEXISTS,
-	wordNOTEXISTS:   parserTokenNOTEXISTS,
-}
-
-var wordToToken = map[string]parserToken{
-	wordAND:   parserTokenAND,
-	wordOR:    parserTokenOR,
-	wordTRUE:  parserTokenTRUE,
-	wordFALSE: parserTokenFALSE,
-}
-
-var runeToToken = map[rune]parserToken{
-	runeEOF:                   parserTokenEOF,
-	runeLEFTPARENTHESE:        parserTokenLEFTPARENTHESE,
-	runeRIGHTPARENTHESE:       parserTokenRIGHTPARENTHESE,
-	runeQUOTE:                 parserTokenQUOTE,
-	runeSINGLEQUOTE:           parserTokenSINGLEQUOTE,
-	runeLEFTSQUAREPARENTHESE:  parserTokenLEFTSQUAREPARENTHESE,
-	runeRIGHTSQUAREPARENTHESE: parserTokenRIGHTSQUAREPARENTHESE,
-	runeCOMMA:                 parserTokenCOMMA,
-}
+var (
+	operatorsToToken = map[string]parserToken{
+		wordEQUAL:       parserTokenEQUAL,
+		wordNOTEQUAL:    parserTokenNOTEQUAL,
+		wordLT:          parserTokenLT,
+		wordLTE:         parserTokenLTE,
+		wordGT:          parserTokenGT,
+		wordGTE:         parserTokenGTE,
+		wordCONTAINS:    parserTokenCONTAINS,
+		wordNOTCONTAINS: parserTokenNOTCONTAINS,
+		wordMATCHES:     parserTokenMATCHES,
+		wordIN:          parserTokenIN,
+		wordNOTIN:       parserTokenNOTIN,
+		wordNOT:         parserTokenNOT,
+		wordEXISTS:      parserTokenEXISTS,
+		wordNOTEXISTS:   parserTokenNOTEXISTS,
+		wordNOTEXISTSSP: parserTokenNOTEXISTS,
+	}
+
+	tokensToOperator = map[parserToken]string{
```

Comment:
we don't really need this, just for the sake of optimizing an error message generation. Let's loop on the original map when needed

Reply:
on it

Reply:
donezo